lib/logfmt/encoder.ex

```diff
@@ -14,40 +51,38 @@ defmodule Logfmt.Encoder do
   @doc """
   See [`Logfmt.encode`](/logfmt/Logfmt.html#encode/1).
   """
-  @spec encode(Dict.t) :: String.t
-  def encode(list) do
-    Enum.reduce list, "", &encode_pair/2
-  end
-
-  @spec encode_pair({Dict.key, String.t | boolean | number}, String.t) :: String.t
-  defp encode_pair({key, value}, "") when is_binary(value) do
-    if String.contains?(value, "=") do
-      ~s(#{key}="#{encode_value(value)}")
-    else
-      ~s(#{key}=#{encode_value(value)})
+  @spec encode(Dict.t, options :: Keyword.t) :: String.t
+  def encode(list, options \\ []) do
+    result = Enum.reduce(list, [], &encode_pair/2)
+    case options[:output] do
+      # in case you wish to get back an iolist
+      :iolist -> result
+      # otherwise default to returning a binary
+      _ -> Enum.join(result)
     end
   end
 
-  defp encode_pair({key, value}, "") do
-    ~s(#{key}=#{encode_value(value)})
+  @spec encode_value(value :: term) :: String.t
+  defp encode_value(value) do
+    str = Logfmt.ValueEncoder.encode(value)
+    cond do
```

Comment:
There are only two branches here, mind refactoring it to an `if`?

Reply:
I was using cond since I intended to check for double or single quotes in the string before wrapping.

I've changed it to an if now